### PR TITLE
Add new job

### PR DESCRIPTION
**Pull request by FadiAlamassi**
* Create hire route to add the job contract to the database
* Validate the received data from the form
* Create a query to add the job contract to the database
* Test hire route
relates #42

### CODE REVIEW

File: server/controllers/ValidationSchemas/newJobSchema.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+const yup = require('yup');
+
+exports.newJobSchema = yup.object({
+  deadline: yup.date().min(new Date()),
+  price: yup
```
Comment (mossa-Sammer) [inline server/controllers/ValidationSchemas/newJobSchema.js]: in one line is better

File: server/controllers/routes/postJob.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+const { postJob } = require('../../db/queries/postJob');
+const { newJobSchema } = require('../ValidationSchemas/newJobSchema');
+
+exports.postJob = async (req, res, next) => {
+  const { clientId } = req.cookies;
+  req.body.clientId = clientId;
+  try {
+    await newJobSchema.validate(req.body);
+    await postJob(req.body);
+    res.set('Content-Type', 'application/json');
```
Comment (mossa-Sammer) [inline server/controllers/routes/postJob.js]: why res.set ?

File: server/controllers/routes/postJob.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+const { postJob } = require('../../db/queries/postJob');
+const { newJobSchema } = require('../ValidationSchemas/newJobSchema');
+
+exports.postJob = async (req, res, next) => {
+  const { clientId } = req.cookies;
```
Comment (mossa-Sammer) [inline server/controllers/routes/postJob.js]: use the unlock cookie the middleware to get the client id

File: server/tests/routes/postJob.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    description: 'build a house',
+    street: 'Tal Elhawa',
+    status: 'pending',
+    buildingNumber: '111',
+    flatNumber: '222',
+  };
+  request(app)
+    .post('/api/v1/hire')
+    .send(jobContract)
+    .set('Accept', 'application/json')
```
Comment (AbdallahAmmar96) [inline server/tests/routes/postJob.js]: delete this line

File: server/controllers/routes/postJob.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+const { postJob } = require('../../db/queries/postJob');
+const { newJobSchema } = require('../ValidationSchemas/newJobSchema');
+
+exports.postJob = async (req, res, next) => {
+  req.body.clientId = req.user.id;
+  try {
+    await newJobSchema.validate(req.body);
+    await postJob(req.body);
+    res.send({ message: 'success', statusCode: 200 });
```
Comment (AbdallahAmmar96) [inline server/controllers/routes/postJob.js]: Job added successfully

File: server/controllers/ValidationSchemas/newJobSchema.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+const yup = require('yup');
+
+exports.newJobSchema = yup.object({
+  deadline: yup.date().min(new Date()),
+  price: yup.number().positive().required().min(5),
```
Comment (AbdallahAmmar96) [inline server/controllers/ValidationSchemas/newJobSchema.js]: change the min and requiered

File: server/controllers/ValidationSchemas/newJobSchema.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+  description: yup.string().min(5).required(),
```
Comment (AbdallahAmmar96) [inline server/controllers/ValidationSchemas/newJobSchema.js]: min(5) ???

**Pull request merged**